packages/components/src/pages/VsCodeExtension.vue

```diff
@@ -487,10 +492,14 @@ export default {
       if (this.filters.rootObjects.length) {
         let eventBranches = [];
 
-        this.filters.rootObjects.forEach((id) => {
-          const filterRegExp = new RegExp(id, 'ig');
-          classMap.codeObjects.forEach((codeObject) => {
-            if (!filterRegExp.test(codeObject.fqid)) {
+        classMap.codeObjects.forEach((codeObject) => {
+          this.filters.rootObjects.forEach((id) => {
+            if (id.includes('*')) {
+              const filterRegExp = new RegExp(id, 'ig');
+              if (!filterRegExp.test(codeObject.fqid)) {
+                return;
+              }
+            } else if (id !== codeObject.fqid) {
```

Comment:
I don't think this is what we want. The regex star operator matches zero or more of the preceding token and doesn't act as a glob. If we intend for this to be a glob, the regex should be generated from the input (e.g. `package:*/controllers` -> `^package:.*/controllers$`)

As written now, if you're looking to match `package:app/controllers`, the root filter `package:*/controllers` will yield no results. There should be a test case that validates this behaves as intended.

Reply:
Looks like I forgot to replace `*` with `.*`.
I've made changes and added new e2e test for `package:*/controllers` filter query.
https://github.com/applandinc/appmap-js/pull/429/commits/656103591f61226dc3b1c04881df6eb14dcccf1e